### PR TITLE
BUG: Empty pipeline failed to tz_localize

### PR DESCRIPTION
**Pull request by dmichalowicz**
`tz_localize` was failing on a dataframe with an empty MultiIndex.

### CODE REVIEW

File: zipline/pipeline/engine.py [hunk 1]
```diff
@@ -384,6 +384,20 @@ def _to_narrow(self, data, mask, dates, assets):
         If mask[date, asset] is True, then result.loc[(date, asset), colname]
         will contain the value of data[colname][date, asset].
         """
+        if not mask.any():
+            # Manually handle the empty DataFrame case. This is a workaround
```
Comment (ssanderson) [inline zipline/pipeline/engine.py]: Might be worth noting that this also saves us the work of applying a known-empty mask to the array.

File: tests/pipeline/test_pipeline_algo.py [hunk 2]
```diff
@@ -566,3 +566,36 @@ def handle_data(context, data):
...
+            pipeline.add(vwap, 'vwap')
+
+            # Nothing should have prices less than 0.
+            pipeline.set_screen(vwap < 0)
+
+        def handle_data(context, data):
+            pass
+
+        def before_trading_start(context, data):
+            context.results = pipeline_output('test')
```
Comment (ssanderson) [inline tests/pipeline/test_pipeline_algo.py]: Can we assert that we get an empty DataFrame here?

Comment (ssanderson) [inline tests/pipeline/test_pipeline_algo.py]: Should probably also assert that BTS actually called here.

**Comment by dmichalowicz** [comment]
Got the okay from @ssanderson

**Pull request merged**